modules/i18n/src/main/java/com/devonfw/module/i18n/common/util/I18nUtils.java

```diff
@@ -0,0 +1,198 @@
+package com.devonfw.module.i18n.common.util;
+
+import java.io.File;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import net.sf.mmm.util.nls.base.ResourceBundleControlUtf8WithNlsBundleSupport;
+
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * Utility class for i18n module.
+ *
+ */
+public class I18nUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(I18nUtils.class);
+
+  /**
+   * @param resource is the bundle of message resources
+   * @return map containing key value pairs of message resources
+   */
+  public static Map<String, String> convertResourceBundleToMap(ResourceBundle resource) {
+
+    Map<String, String> map = new HashMap<>();
+
+    Enumeration<String> keys = resource.getKeys();
+    while (keys.hasMoreElements()) {
+      String key = keys.nextElement();
+      map.put(key, resource.getString(key));
+    }
+    return map;
+  }
+
+  /**
+   * @param objJSON JSON String which is the representation of key value pairs in message resources property file
+   * @param filter passed from the service
+   * @param strJSONKey JSON Key
+   * @param count used to traverse the JSON
+   * @return JSON String that is matched against the filter
+   * @throws JSONException thrown by getWithDotNotation
+   */
+  public static String getWithDotNotation(JsonObject objJSON, String filter, String strJSONKey, int count)
+      throws JSONException {
+
+    String closingBraces = I18nConstants.EMPTY_STRING;
+    String strJSON = I18nConstants.EMPTY_STRING;
+    if (filter.contains(I18nConstants.DOT)) {
+      int indexOfDot = filter.indexOf(I18nConstants.DOT);
+      String subKey = filter.substring(0, indexOfDot);
+      JsonObject jsonObject = (JsonObject) objJSON.get(subKey);
+      count++;
+      strJSONKey = strJSONKey + "{\"" + subKey + "\":";
```

Comment:
It also seems an anti-pattern to me to generate JSON via "string-fiddling".
Is maybe gson the wrong choice here? I have not used it as there is a JEE standard JSON-P for it and we comitted to use JEE standards where suitable instead of using proprietary solutions:
https://github.com/devonfw/devon4j/blob/develop/documentation/guide-jee.asciidoc#jee